hazelcast/src/main/java/com/hazelcast/map/impl/PartitioningStrategyFactory.java

```diff
@@ -47,31 +48,37 @@ private PartitioningStrategyFactory() {
      */
     public static PartitioningStrategy getPartitioningStrategy(NodeEngine nodeEngine, String mapName,
                                                                PartitioningStrategyConfig config) {
-            PartitioningStrategy strategy = null;
-            if (config != null) {
-                strategy = config.getPartitioningStrategy();
-                if (strategy == null) {
-                    if (CACHE.containsKey(mapName)) {
-                        strategy = CACHE.get(mapName);
-                    } else if (config.getPartitioningStrategyClass() != null) {
-                        try {
-                            strategy = ClassLoaderUtil.newInstance(nodeEngine.getConfigClassLoader(),
-                                    config.getPartitioningStrategyClass());
-                            CACHE.put(mapName, strategy);
-                        } catch (Exception e) {
-                            throw ExceptionUtil.rethrow(e);
-                        }
+        String key = key(nodeEngine, mapName);
+        PartitioningStrategy strategy = null;
+        if (config != null) {
+            strategy = config.getPartitioningStrategy();
+            if (strategy == null) {
+                if (CACHE.containsKey(key)) {
+                    strategy = CACHE.get(key);
+                } else if (config.getPartitioningStrategyClass() != null) {
+                    try {
+                        strategy = ClassLoaderUtil
+                                .newInstance(nodeEngine.getConfigClassLoader(), config.getPartitioningStrategyClass());
+                        CACHE.put(key, strategy);
+                    } catch (Exception e) {
+                        throw ExceptionUtil.rethrow(e);
                     }
                 }
             }
-            return strategy;
+        }
+        return strategy;
     }
 
     /**
      * Remove the cached {@code PartitioningStrategy} from the internal cache, if it exists.
+     * @param nodeEngine the NodeEngine in which {@code mapName} is defined
      * @param mapName name of the map whose partitioning strategy will be removed from internal cache
      */
-    public static void removePartitioningStrategyFromCache(String mapName) {
-        CACHE.remove(mapName);
+    public static void removePartitioningStrategyFromCache(NodeEngine nodeEngine, String mapName) {
+        CACHE.remove(key(nodeEngine, mapName));
+    }
+
+    static String key(NodeEngine nodeEngine, String mapName) {
+        return nodeEngine.getLocalMember().getUuid() + ":" + mapName;
```

Comment:
Compose cache key using local member's UUID & `mapName` so that we don't clash with other nodes in the same JVM.